.github/workflows/ci.yml

```diff
@@ -0,0 +1,145 @@
+          ./run-tests.sh --check-shellscript
+
+  Lint-black:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Check Python code formatting
+        run: |
+          pip install black
+          ./run-tests.sh --check-black
+
+  Lint-pydocstyle:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Check compliance with Python docstring conventions
+        run: |
+          pip install pydocstyle
+          ./run-tests.sh --check-pydocstyle
+
+  Lint-check-manifest:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Check Python manifest completeness
+        run: |
+          pip install check-manifest
+          ./run-tests.sh --check-manifest
+
+  Docs-sphinx:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install system dependencies
+        run: |
+          sudo apt-get update -y
+          sudo apt install libcurl4-openssl-dev libssl-dev
+          sudo apt-get install libgnutls28-dev
+
+      - name: Install Python dependencies
+        run: |
+          pip install --upgrade pip setuptools py
+          pip install -e .[all]
+
+      - name: Run Sphinx documentation with doctests
+        run: ./run-tests.sh --check-sphinx
+
+  Python-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Python dependencies
+        run: |
+          pip install --upgrade pip setuptools py
+          pip install twine wheel
+          pip install -e .[all]
+
+      - name: Run pytest
+        run: ./run-tests.sh
+
+      - name: Codecov Coverage
+        if: matrix.python-version == 3.8
+        uses: codecov/codecov-action@v1
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: coverage.xml
+
+  Lint-helm:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Fetch history
+        run: git fetch --prune --unshallow
+
+      - name: Run chart-testing (lint)
+        id: lint
+        uses: helm/chart-testing-action@v1.1.0
```

Comment:
After skimming through this action docs, I see it uses [kind-action](https://github.com/helm/kind-action). We could use it to do nightly builds inside this repo [using cron job style in GH Actions](https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#scheduled-events). CC @tiborsimko if this would make sense we could create a ticket.

Reply:
It might be good if resources allow... Let's recall it for later.